controllers/clustermesh/clustermesh_controller.go

```diff
@@ -123,63 +195,125 @@ func (r *ClusterMeshReconciler) reconcileDelete(ctx context.Context, cluster *cl
 
 	if len(clustermesh.Spec.Clusters) == 0 {
 		if err := r.Client.Delete(ctx, clustermesh); err != nil {
-			return ctrl.Result{}, err
+			return err
 		}
-		return ctrl.Result{}, nil
+		return nil
 	} else {
 		if err := r.Client.Update(ctx, clustermesh); err != nil {
-			return ctrl.Result{}, err
+			return err
 		}
 	}
 	return r.ReconcilePeeringsFactory(r, ctx, clustermesh)
 }
 
-func (r *ClusterMeshReconciler) reconcileNormal(ctx context.Context, cluster *clusterv1beta1.Cluster, clustermesh *clustermeshv1beta1.ClusterMesh) (ctrl.Result, error) {
-	clSpec, err := r.PopulateClusterSpecFactory(r, ctx, cluster)
+func ReconcileSecurityGroups(r *ClusterMeshReconciler, ctx context.Context, clustermesh *clustermeshv1beta1.ClusterMesh) error {
+	// Update status
+	ownedSecurityGroupsRefs, err := crossplane.GetOwnedSecurityGroups(ctx, clustermesh, r.Client)
 	if err != nil {
-		return ctrl.Result{}, err
+		return err
 	}
+	clustermesh.Status.CrossplaneSecurityGroupRef = ownedSecurityGroupsRefs
 
-	key := client.ObjectKey{
-		Name: cluster.Labels["clusterGroup"],
+	sgPrefix := "clustermesh-"
+	cidrResults := map[string][]string{}
+	for _, cl := range clustermesh.Spec.Clusters {
```

Comment:
We already use the kcp to retrieve the region and subnet of the Clusters before. WDYT about also retrieving the CIDR to avoid having to do this again?